src/System.Private.CoreLib/shared/System/SpanHelpers.Char.cs

```diff
@@ -218,93 +221,202 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
         {
             Debug.Assert(length >= 0);
 
-            fixed (char* pChars = &searchSpace)
-            {
-                char* pCh = pChars;
-                char* pEndCh = pCh + length;
+            nint offset = (nint)0;
+            nint lengthToExamine = length;
 
-                if (Vector.IsHardwareAccelerated && length >= Vector<ushort>.Count * 2)
+            if (Avx2.IsSupported || Sse2.IsSupported)
```

Comment:
Why not just `Sse2.IsSupported`?

Reply:
Clarity? Is used for both Avx2 and Sse2 paths

Reply:
A comment might be better?

You can only have one of the following states:
* SSE2 = false, AVX2 = false
* SSE2 = true, AVX2 = false
* SSE2 = true, AVX2 = true

So the `AVX2.IsSupported` check is unnecessary